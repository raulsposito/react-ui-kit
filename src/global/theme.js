// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
