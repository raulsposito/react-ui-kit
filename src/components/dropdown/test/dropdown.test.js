import React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { colors, spacing } from '@global/theme'

import Dropdown from '..'

const mockData = {
  placeholder: 'Dropdown Placeholder',
  onChange: jest.fn()
}

let selectedOption = {
  value: 'apartamento',
  label: 'Apartamento',
  icon: 'apartment'
}

const handleChage = e => { selectedOption = e }

const mockOptions = [
  { value: 'tienda', label: 'Tienda', icon: 'storefront' },
  { value: 'gimnasio', label: 'Gimnasio' },
  { value: 'bar', label: 'Bar', icon: 'sports_bar' },
  { value: 'spa', label: 'Spa', icon: 'spa' },
  { value: 'casa', label: 'Casa' },
  { value: 'playa', label: 'Playa' },
  { value: 'apartamento', label: 'Apartamento', icon: 'apartment' },
  { value: 'hotel', label: 'Hotel' }
]

describe('<Dropdown />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  describe('Placeholder', () => {
    it('input has placeholder', () => {
      render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

      const placeholder = screen.queryByText(mockData.placeholder)
      expect(placeholder).toBeInTheDocument()
    })
  })

  describe('Input event', () => {
    it('calls function on change', () => {
      render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')

      userEvent.click(dropdownControl)
      const firstOption = screen.queryByText(mockOptions[1].label).closest('div')

      userEvent.click(firstOption)
      expect(mockData.onChange).toHaveBeenCalledTimes(1)
    })
  })

  describe('Variant Line', () => {
    it('is regular', () => {
      render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      const dropdownValueContainer = screen.queryByText(mockData.placeholder).parentElement
      const dropdownPlaceholder = screen.queryByText(mockData.placeholder).closest('div')
      const dropdownIndicatorContainer = screen.queryByText('arrow_drop_down').closest('div')
      const dropdownIndicator = screen.queryByText('arrow_drop_down')

      expect(dropdownControl).toHaveStyle({ border: '2px solid' })
      expect(dropdownControl).toHaveStyle({ borderRadius: '5px' })
      expect(dropdownControl).toHaveStyle({ boxSizing: 'border-box' })
      expect(dropdownControl).toHaveStyle({ boxShadow: 'none' })
      expect(dropdownControl).toHaveStyle({ cursor: 'pointer' })
      expect(dropdownControl).toHaveStyle({ color: `${colors.primary.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.background.regular}` })

      expect(dropdownValueContainer).toHaveStyle({ paddingLeft: `${spacing.two}` })

      expect(dropdownPlaceholder).toHaveStyle({ color: 'inherit' })

      expect(dropdownIndicatorContainer).toHaveStyle({ paddingRight: `${spacing.two}` })

      expect(dropdownIndicator).toHaveStyle({ color: `${colors.system.regular}` })
      expect(dropdownIndicator).toBeInTheDocument()
    })

    it('is primary', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='primary'
        />
      )
      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.primary.regular}` })
    })

    it('is secondary', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='secondary'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.secondary.regular}` })
    })

    it('is variant', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='variant'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.variant.regular}` })
    })

    it('is success', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='success'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.success.regular}` })
    })

    it('is error', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='error'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.error.regular}` })
    })

    it('is system', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='system'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.system.regular}` })
    })
  })

  describe('Variant Solid', () => {
    it('is regular', () => {
      render(
        <Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} type='solid' />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      const dropdownValueContainer = screen.queryByText(mockData.placeholder).parentElement
      const dropdownPlaceholder = screen.queryByText(mockData.placeholder).closest('div')
      const dropdownIndicatorContainer = screen.queryByText('arrow_drop_down').closest('div')
      const dropdownIndicator = screen.queryByText('arrow_drop_down')

      expect(dropdownControl).toHaveStyle({ border: '2px solid' })
      expect(dropdownControl).toHaveStyle({ borderRadius: '5px' })
      expect(dropdownControl).toHaveStyle({ boxSizing: 'border-box' })
      expect(dropdownControl).toHaveStyle({ boxShadow: 'none' })
      expect(dropdownControl).toHaveStyle({ cursor: 'pointer' })
      expect(dropdownControl).toHaveStyle({ color: `${colors.primary.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.primary.regular}` })

      expect(dropdownValueContainer).toHaveStyle({ paddingLeft: `${spacing.two}` })

      expect(dropdownPlaceholder).toHaveStyle({ color: `${colors.events.white}` })

      expect(dropdownIndicatorContainer).toHaveStyle({ paddingRight: `${spacing.two}` })

      expect(dropdownIndicator).toHaveStyle({ color: `${colors.background.regular}` })
      expect(dropdownIndicator).toBeInTheDocument()
    })

    it('is primary', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='primary'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.primary.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.primary.regular}` })
    })

    it('is secondary', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='secondary'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.secondary.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.secondary.regular}` })
    })

    it('is variant', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='variant'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.variant.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.variant.regular}` })
    })

    it('is success', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='success'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.success.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.success.regular}` })
    })

    it('is error', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='error'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.error.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.error.regular}` })
    })

    it('is system', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          onChange={mockData.onChange}
          type='solid'
          color='system'
        />
      )

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      expect(dropdownControl).toHaveStyle({ color: `${colors.system.regular}` })
      expect(dropdownControl).toHaveStyle({ backgroundColor: `${colors.system.regular}` })
    })
  })

  describe('Dropdown Options', () => {
    it('regular', () => {
      render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      userEvent.click(dropdownControl)

      const firstOption = screen.queryByText(mockOptions[1].label).closest('div')

      expect(firstOption).toHaveStyle({ display: 'flex' })
      expect(firstOption).toHaveStyle({ alignItems: 'center' })
      expect(firstOption).toHaveStyle({ justifyContent: 'space-between' })
      expect(firstOption).toHaveStyle({ padding: `${spacing.one} ${spacing.two} ${spacing.one} ${spacing.two}` })
      expect(firstOption).toHaveStyle({ backgroundColor: `${colors.background.primary}` })
      expect(firstOption).toHaveStyle({ color: `${colors.contrast.regular}` })
    })

    it('hover', () => {
      render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

      const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
      userEvent.click(dropdownControl)

      const firstOption = screen.queryByText(mockOptions[1].label).closest('div')
      userEvent.hover(firstOption)

      expect(firstOption).toHaveStyle({ backgroundColor: `${colors.background.hover}` })
      expect(firstOption).toHaveStyle({ color: `${colors.primary.regular}` })
    })

    describe('Icons Option', () => {
      it('has icon', () => {
        render(<Dropdown options={mockOptions} placeholder={mockData.placeholder} onChange={mockData.onChange} />)

        const dropdownControl = screen.queryByText(mockData.placeholder).parentElement.parentElement.closest('div')
        userEvent.click(dropdownControl)

        const firstOption = screen.queryByText(mockOptions[1].label).closest('div')
        expect(firstOption).toBeInTheDocument()
      })
    })
  })

  describe('Controlled Component Behavior', () => {
    it('controlled value set', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          value={selectedOption}
          onChange={handleChage}
        />
      )

      const dropdownControlled = screen.queryByText(selectedOption.label).closest('div')
      expect(dropdownControlled).toBeInTheDocument()
    })

    it('controlled value change', () => {
      render(
        <Dropdown
          options={mockOptions}
          placeholder={mockData.placeholder}
          value={selectedOption}
          onChange={handleChage}
        />
      )

      const dropdownControlled = screen.queryByText(selectedOption.label).closest('div')
      userEvent.click(dropdownControlled)

      const firstOption = screen.queryByText(mockOptions[1].label).closest('div')
      expect(firstOption).toBeInTheDocument()
      userEvent.click(firstOption)

      const changedDropdownValue = screen.queryByText(selectedOption.label)
      waitFor(() => expect(changedDropdownValue).toBeInTheDocument())
    })
  })
})
