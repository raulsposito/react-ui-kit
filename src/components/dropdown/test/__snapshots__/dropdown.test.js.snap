// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTEea iQNNfU css-dyr6gj-container"
  >
    <span
      aria-atomic="false"
      aria-live="polite"
      aria-relevant="additions text"
      class="css-1f43avz-a11yText-A11yText"
    />
    <div
      class=" css-1bpxup0-control"
    >
      <div
        class=" css-x1hhe8-ValueContainer"
      >
        <div
          class=" css-ogw9ai-placeholder"
        >
          Dropdown Placeholder
        </div>
        <input
          aria-autocomplete="list"
          class="css-wmatm6-dummyInput-DummyInput"
          id="react-select-2-input"
          readonly=""
          tabindex="0"
          value=""
        />
      </div>
      <div
        class=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          class=" css-43ykx9-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          class=" css-1e4t2s-indicatorContainer"
        >
          <span
            class="sc-bdfBQB VgOlj material-icons"
            color="system"
          >
            arrow_drop_down
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
